### PR TITLE
TAS-104/Supprimer_Credit_Card

### PR DESCRIPTION
**Pull request by Brendan-bx**

### CODE REVIEW

File: apps/api/src/routes/users/creditCards.ts [hunk 1]
```diff
@@ -9,28 +9,51 @@ import {
...
+					)
+				}
+			},
+		)
+		.delete(
+			"/:cardId",
+			zValidator("param", z.object({ cardId: idValidator })),
+			async (c) => {
+				const { cardId } = c.req.valid("param")
+				const { prisma } = c.var
```
Comment (Limerio) [inline apps/api/src/routes/users/creditCards.ts]: how can I know if a credit card doesn't exist? Because you don't checking it

File: apps/api/src/user.js
Comment (Limerio) [inline apps/api/src/user.js:1]: remove this script

**Review by Limerio — CHANGES_REQUESTED** [review]

File: apps/api/src/routes/users/creditCards.ts [hunk 2]
```diff
@@ -9,32 +9,66 @@ import { Hono } from "hono"
...
-			const { userId } = c.req.valid("param")
-			const { prisma } = c.var
-			const data = c.req.valid("json")
+	new Hono<{ Variables: PrivateContextVariables }>()
+		.post(
+			"/",
+			zValidator("param", z.object({ userId: idValidator })),
+			zValidator("json", creditCardValidator),
+			async (c) => {
+				const { userId } = c.req.valid("param")
```
Comment (WookiesRpeople2) [inline apps/api/src/routes/users/creditCards.ts]: you do not have any parms here have you tested this code

Comment (WookiesRpeople2) [reply apps/api/src/routes/users/creditCards.ts]:
```suggestion
.post(
			"/",
			zValidator("param", z.object({ userId: idValidator })),
			zValidator("json", creditCardValidator),
			async (c) => {
```

File: apps/api/src/routes/users/creditCards.ts [hunk 3]
```diff
@@ -9,32 +9,66 @@ import { Hono } from "hono"
...
-				const newCreditCard = await prisma.creditCard.create({
-					data: {
-						...data,
-						userId,
-					},
-				})
+				try {
+					const newCreditCard = await prisma.creditCard.create({
+						data: {
+							...data,
```
Comment (WookiesRpeople2) [inline apps/api/src/routes/users/creditCards.ts:24]:
```suggestion
const newCreditCard = await prisma.creditCard.create({
						data: {
							data,
						},
					})
					
```

File: apps/api/src/routes/users/creditCards.ts (hunk 1 above)
Comment (WookiesRpeople2) [reply apps/api/src/routes/users/creditCards.ts]: look into the function ```findfirst``` and ```findunique```

File: apps/api/src/routes/users/creditCards.ts [hunk 4]
```diff
@@ -9,32 +9,66 @@ import { Hono } from "hono"
...
+					)
+				}
+			},
+		)
+		.delete(
+			"/:cardId",
+			zValidator("param", z.object({ cardId: idValidator })),
+			async (c) => {
+				const { cardId } = c.req.valid("param")
+				const { prisma } = c.var
```
Comment (WookiesRpeople2) [inline apps/api/src/routes/users/creditCards.ts]: destructure prisma dicrectly is nicer to look at

not ```const { prisma } = c.var```

but 
```suggestion

async ({prisma}) => {
				const { cardId } = c.req.valid("param")
				````

**Review by WookiesRpeople2 — CHANGES_REQUESTED** [review]

**Review by WookiesRpeople2 — DISMISSED** [review]

**Review by Limerio — APPROVED** [review]

**Pull request merged**